components/suggestion/switch_channel_provider.jsx

```diff
@@ -332,14 +332,30 @@ export default class SwitchChannelProvider extends Provider {
     }
 
     userWrappedChannel(user, channel) {
-        let displayName = `@${user.username}`;
-
-        if ((user.first_name || user.last_name) && user.nickname) {
-            displayName += ` - ${Utils.getFullName(user)} (${user.nickname})`;
-        } else if (user.nickname) {
-            displayName += ` - (${user.nickname})`;
-        } else if (user.first_name || user.last_name) {
-            displayName += ` - ${Utils.getFullName(user)}`;
+        const teammateNameDisplay = getTeammateNameDisplaySetting(getState());
```

Comment:
Are we sure that we want to change what's displayed in the picker based on the name format? I don't think we do this anywhere else that we autocomplete users.

Reply:
Yeah, it's needed for the autocomplete to give correct results. The problem, as I outlined earlier is that GM names already respect the preference setting. But DMs don't. So if you search for "Harrison", everything is fine, but the moment you enter "Harrison H", GMs jump higher than DMs because DMs have "harrison.healy" before so it doesn't match.

You are right, and I just found that the search bar also has the same problem. We need to update that too.

Reply:
Ok, I checked a bit more and there are 3 places where we need to update this - UserSuggestion, SearchUserSuggestion, and AtMentionSuggestion. Combined with this change, it would change 4 places. Are we sure we want to do this? The initial ticket was meant as a bug in list ordering, but then it actually didn't turn out to be a bug, but a feature change. Now this is turning out to be an even bigger change. The only possible bug here is that `channelTypeOrder` in mattermost-redux is the same for GM and DM whereas DMs should be higher. But that isn't strictly related to this ticket.

@esethna / @wiersgallak - requesting confirmation on whether we really want to go ahead with this.